### ISSUE TITLE
Implement mDCv chunk described in PNG 3rd Edition (Priority 2)

### ISSUE THREAD

**Issue by digitaltvguy** [glennrp/libpng#509]
mDCv or Mastering Display Color Volume (SMPTE ST 2086) stores static metadata which idenfies display characteristics of the mastering display and allows optimized tone and color mapping to a target (consumer display).

mDCv is helpful because the content creation display may have superior capabilities compared to the target display, therefore tone and color mapping may be necessary

**Comment by svgeesus** [comment]
https://w3c.github.io/png/#mDCv-chunk

**Comment by jbowler** [comment]
Yet another broken link :-)  The current **draft** spec is here:

https://www.w3.org/TR/png-3/#mDCv-chunk

The spec is, well, very wordy and not very useful; it's not a normative spec, it's pretty much a discussion.  I'm sure this will change during review; proffering opinions in ISO specs is not normally acceptable!  At present the spec is unimplementable simply because it does not define the encoding of the fields.  It says how many bytes each contains but not how the bytes are encoded!

It's possible to deduce that the CIExyY values are encoded in two bytes for x and y but it is not possible to deduce whether that is a signed or unsigned value given the error in cHRM (it erroneously uses unsigned values).  The ACES AP0 color space has a negative "y" (and, as discussed, AP1 has a negative "z" but that's ok, as discussed.)

The same problem may exist with cLLi because display relative images can, in fact, have negative luminance.  This is a consequence of the need to offset colours to make them visible, so in display space a colour may end up with a negative luminance because it is below the level of perception in the viewing environment (the "black level").

**Comment by digitaltvguy** [comment]
Negative luminance values in narrow range can also contain pluge (which allows monitor setup for black levels)

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: John Bowler ***@***.***>
Sent: Friday, September 13, 2024 11:28:30 PM
To: pnggroup/libpng ***@***.***>
Cc: Seeger, Chris (NBCUniversal) ***@***.***>; Author ***@***.***>
Subject: [EXTERNAL] Re: [pnggroup/libpng] Implement mDCv chunk described in PNG 3rd Edition (Priority 2) (Issue #509)


Yet another broken link :-) The current draft spec is here:

https://www.w3.org/TR/png-3/#mDCv-chunk<https://urldefense.com/v3/__https://www.w3.org/TR/png-3/*mDCv-chunk__;Iw!!PIZeeW5wscynRQ!o4a0hdot9Sz7xQWBGhep60fgNMq9nS2MR3t58Kkd0jgpPG4-I6eXJHqQTwSR7Y6gKpxXzO-xSsPrWVwNPiO6ZQzGUg$>

The spec is, well, very wordy and not very useful; it's not a normative spec, it's pretty much a discussion. I'm sure this will change during review; proffering opinions in ISO specs is not normally acceptable! At present the spec is unimplementable simply because it does not define the encoding of the fields. It says how many bytes each contains but not how the bytes are encoded!

It's possible to deduce that the CIExyY values are encoded in two bytes for x and y but it is not possible to deduce whether that is a signed or unsigned value given the error in cHRM (it erroneously uses unsigned values). The ACES AP0 color space has a negative "y" (and, as discussed, AP1 has a negative "z" but that's ok, as discussed.)

The same problem may exist with cLLi because display relative images can, in fact, have negative luminance. This is a consequence of the need to offset colours to make them visible, so in display space a colour may end up with a negative luminance because it is below the level of perception in the viewing environment (the "black level").

**Comment by jbowler** [comment]
> Negative luminance values in narrow range can also contain pluge (which allows monitor setup for black levels)

Indeed, and I would expect a display application or editor which handles the cICP chunk to handle values outside the 16..235 range.  Indeed any app that reads Kodak's PhotoCD already has to handle the negative values.  PhotoCD is a 16..235 encoding, though (IRC) without the signaling values.

Similarly image resampling produces RGB values outside the bounds of the original values even given perfectly sampled image data and perfect resampling (sinc, done correctly.)

**Comment by jbowler** [comment]
Another note; the current spec defines the fields thus:

https://www.w3.org/TR/png-3/#mDCv-chunk-syntax

To quote the table (table 19 at present):

|Name|Size|Divisor value|
|-|-|-|
|Mastering display color primary chromaticities (CIE 1931 x,y of R,G,B )|12 bytes|0.00002|
|Mastering display white point chromaticity (CIE 1931 x,y)|4 bytes|0.00002|
|Mastering display maximum luminance|4 bytes|0.0001 cd/m2|
|Mastering display minimum luminance|4 bytes|0.0001 cd/m2|

The title line of the third column is wrong (the values are scale factors; multipliers not divisors).  That's clear to me; for one thing the last two rows are dimensionally incorrect (luminance is measured in cd/m2, not 1/(cd/m2) but it will probably lead to some confusion.

**Comment by digitaltvguy** [comment]
The standard SMPTE ST 2067-21 indicates them as "units of" in Annex B.  I think all that needs to occur is moving cd/m2 to the name column.  The store value is multiplied by the divisor to get the resulting luminance in cd/m2

**Comment by digitaltvguy** [comment]
Change has been posted to branch 464 in PNG 3rd Edition

**Comment by jbowler** [comment]
Ok.  It would be useful to know the maximum values of all four fields; it's no problem with floating point but libpng also has fixed point APIs.

**Comment by ProgramMax** [comment]
Following up here, the mDCv chunk does specify the byte definition. So it can be implemented.

Although negative luminance values in narrow range can act as a control signal, those signal values are separate from the display's color volume. I think some of the conversation here is mixing logical color spaces with the physical mastering display's color volume, right?

**Comment by jbowler** [comment]
ACES is intended for use in video workflows; it was originated (in 2014) by the film industry.  It is a mastering color space.  Clarification is required in the spec with regard to specifications like ACES.  Are we meant to put the workflow space in cHRM (etc) or in mDCv?

**Comment by ProgramMax** [comment]
"Mastering Display" here is the monitor/display used while mastering.
That is separate from the image's color space. And the workflow color space could also be different.

It would be possible to use ACES in the workflow and then render out to a final, non-ACES color space for the end user.

The reason we care about the mastering display at all is for the end user's processing to try to match it. It is a way to provide a target for the end device.

**Comment by digitaltvguy** [comment]
Hi John,

Mdcv requires a display-referred system so that there is a definitive luminance value.

ACES is a scene-referred and is therefore represents relative levels:

https://www.toodee.de/using-aces/display-referred-vs-scene-referred/

I don’t see how a relative system can represent absolute values that are required by MDCV (which describes a displays min/max luminance level capabilities).

MDCV describes the technical characteristics of the “Mastering Display Color Volume” itself.  It is not a relative “mastering color space”.

The png working color space for video workflows would be described in CICP where the color primaries, transfer function (light mapping), matrix coefficient and signal range(quantization in full or narrow range) are described.

Best,
Chris


From: John Bowler ***@***.***>
Date: Sunday, September 29, 2024 at 2:26 PM
To: pnggroup/libpng ***@***.***>
Cc: Seeger, Chris (NBCUniversal) ***@***.***>, Author ***@***.***>
Subject: [EXTERNAL] Re: [pnggroup/libpng] Implement mDCv chunk described in PNG 3rd Edition (Priority 2) (Issue #509)

ACES is intended for use in video workflows; it was originated (in 2014) by the film industry. It is a mastering color space. Clarification is required in the spec with regard to specifications like ACES. Are we meant to put the workflow space in cHRM (etc) or in mDCv?

**Comment by digitaltvguy** [comment]
It is useful if the content was mastered with a display that had a higher luminance then a target display so a tone-mapping can be built to avoid simple clipping.

…very useful for HDR display or when remapping HDR to SDR…


Best,
Chris

From: Chris Blume ***@***.***>
Date: Sunday, September 29, 2024 at 2:44 PM
To: pnggroup/libpng ***@***.***>
Cc: Seeger, Chris (NBCUniversal) ***@***.***>, Author ***@***.***>
Subject: [EXTERNAL] Re: [pnggroup/libpng] Implement mDCv chunk described in PNG 3rd Edition (Priority 2) (Issue #509)

"Mastering Display" here is the monitor/display used while mastering.
That is separate from the image's color space. And the workflow color space could also be different.

It would be possible to use ACES in the workflow and then render out to a final, non-ACES color space for the end user.

The reason we care about the mastering display at all is for the end user's processing to try to match it. It is a way to provide a target for the end device.

**Comment by jbowler** [comment]
Fixed by #635; this is ready to test too.  See  https://github.com/jbowler/libpng/tree/pngv3-mDCV%2BcLLI

**Comment by jbowler** [comment]
@digitaltvguy I added better handling of mDCV in #645 but the basic fix is in 1.6.46.  Perhaps you can verify and close this issue?

Comments on the implementation are, of course, welcome.

**Comment by svgeesus** [comment]
Closed by https://github.com/pnggroup/libpng/pull/645
